src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts

```diff
@@ -0,0 +1,287 @@
+    let _positives = [...positives];
+    while (_negatives.length + _positives.length > ticksNumber) {
+        if (_negatives.length >= _positives.length) {
+            _negatives = tail(_negatives);
+        } else {
+            _positives = initial(_positives);
+        }
+    }
+    return [_negatives, _positives];
+}
+
+/**
+ * create the range where zero is between start and end
+ * @param start
+ * @param end
+ * @param ticksNumber
+ */
+function createArrayFromRangeWithMiddleZero(start: number, end: number, ticksNumber: number) {
+    const delta = Math.abs((end - start) / (ticksNumber - 1));
+    const [negatives, positives] = handleOutOfRange(
+        getRange(start, delta),
+        getRange(end, delta),
+        ticksNumber
+    );
+
+    return negatives.concat([0], positives);
+}
+
+/**
+ * Get range from negative to zero and zero to positive
+ * Negatives: start from zero down to target number, value increased by delta
+ * Positives: start from zero up to target number, value increased by delta
+ * @param targetNumber
+ * @param delta
+ */
+function getRange(targetNumber: number, delta: number) {
+    let position = 0;
+    const result = [];
+
+    while (position < Math.abs(targetNumber)) {
+        position += delta;
+        result.push(position);
+    }
+
+    if (targetNumber >= 0) {
+        return result;
+    }
+
+    return result.reverse().map((position: number) => {
+        return 0 - position;
+    });
+}
+
+function maxminCanon(maxmin: any) {
+    const canon: any = [];
+    const extremes = ['min', 'max'];
+
+    maxmin.forEach((item: any, i: number) => {
+        canon[i] = {};
+        extremes.forEach((extreme: string) => {
+            if (item[extreme] === 0) {
+                canon[i][extreme] = 0;
+            } else if (item[extreme] > 0) {
+                canon[i][extreme] = 1;
+            } else {
+                canon[i][extreme] = -1;
+            }
+        });
+    });
+
+    return canon;
+}
+
+/**
+ * New min is calculated base on axis having smallest min
+ * @param maxmin
+ * @param maxminLookup
+ * @param axisIndex
+ */
+function getNewMin(maxmin: any, maxminLookup: any, axisIndex: number) {
+    const idx = (maxmin[0].min <= maxmin[1].min) ? 0 : 1;
+    const f = !maxmin[idx].max ?
+                    maxmin[idx].min : // handle divide zero case
+                    maxmin[idx].min / maxmin[idx].max;
+    return f * maxminLookup[axisIndex].max;
+}
+
+/**
+ * New max is calculated base on axis having largest max
+ * @param maxmin
+ * @param maxminLookup
+ * @param axisIndex
+ */
+function getNewMax(maxmin: any, maxminLookup: any, axisIndex: number) {
+    const idx = (maxmin[0].max > maxmin[1].max) ? 0 : 1;
+    const f = !maxmin[idx].min ?
+                    maxmin[idx].max : // handle divide zero case
+                    maxmin[idx].max / maxmin[idx].min;
+    return f * maxminLookup[axisIndex].min;
+}
+
+function getTickPositioner() {
+    let maxminLookup: any;
```

Comment:
please no more 'any' types. These types are build only here (don't come from Highcharts) so can be defined